File: server/service/APIRequests.py
# ! Importando bibliotecas para requests
import requests

# ! Importando urls base
from utils.Credentials import Credentials as credentials

URLBASE = credentials.get_api_microsservice_base()
SERVICE_LOGIN = credentials.get_service_login()
SERVICE_PWD = credentials.get_service_pwd()


class APIRequests:
    @staticmethod
    async def send_email(to: str, subject: str, text: str):
        # * Definindo url
        url = f"{URLBASE}/email/send"

        # * Fazendo requisição
        resposta = requests.post(
            url,
            json={"to": to, "subject": subject, "text": text},
            headers={"service-login": SERVICE_LOGIN, "service-pwd": SERVICE_PWD},
        )

        # * Verificando se foi bem-sucedido
        if resposta.status_code == 200:
            return True
        else:
            return False

    @staticmethod
    async def request_mining(environment_id: str, repos: list[str]):
        # * Definindo url
        url = f"{URLBASE}/github/mining/repos"
        print(url)
        # * Fazendo requisição
        resposta = requests.post(
            url,
            json={"environment_id": environment_id, "repos": repos},
            headers={"service-login": SERVICE_LOGIN, "service-pwd": SERVICE_PWD},
        )

        # * Verificando se foi bem-sucedido
        if resposta.status_code == 200:
            return True
        else:
            return False

File: server-js/service/APIRequests.js
const config = require("../config.js");
const axios = require("axios");

class APIRequests {
  /**
   * Sends an email using the specified parameters.
   * @param {string} to - The recipient's email address.
   * @param {string} subject - The subject of the email.
   * @param {string} text - The content of the email.
   * @returns {boolean} - True if the email was sent successfully, or false otherwise.
   */
  static async sendEmail(to, subject, text) {
    // * Defining url
    const url = `${config.apiMicroserviceBase}/email/send`;

    // * Making request
    try {
      await axios.post(
        url,
        { to: to, subject: subject, text: text },
        {
          headers: {
            "service-login": config.servicesLogin,
            "service-pwd": config.servicesPassword,
          },
        }
      );
      return true;
    } catch (e) {
      return false;
    }
  }

  /**
   * Makes a request to the mining API endpoint.
   * @param {string} environmentId - The environment ID.
   * @param {Array<string>} repos - The list of repositories.
   * @returns {boolean} - True if the request is successful, or false otherwise.
   */
  static async requestMining(environmentId, repos) {
    // * Defining url
    const url = `${config.apiMicroserviceBase}/github/mining/repos`;

    // * Making request
    try {
      await axios.post(
        url,
        { environment_id: environmentId, repos: repos },
        {
          headers: {
            "service-login": config.servicesLogin,
            "service-pwd": config.servicesPassword,
          },
        }
      );
      return true;
    } catch (e) {
      return false;
    }
  }

  /**
   * Makes a request to the topics API endpoint.
   * @param {string} environmentId - The environment ID.
   * @returns {boolean} - True if the request is successful, or false otherwise.
   */
  static async requestTopics(environmentId) {
    // * Defining url
    const url = `${config.apiMicroserviceBase}/request/topics`;

    // * Making request
    try {
      await axios.post(
        url,
        { environment_id: environmentId },
        {
          headers: {
            "service-login": config.servicesLogin,
            "service-pwd": config.servicesPassword,
          },
        }
      );
      return true;
    } catch (e) {
      return false;
    }
  }
}

module.exports = APIRequests;
